tests/reports/test_reports.py

```diff
@@ -0,0 +1,233 @@
+                                        "fileOwner": "johndoe"
+                                    },
+                                    {
+                                        "snippet": "Name: This is medical advice. Please go and check the doctor.\nEmail: None\nSSN: None\nAddress: None\nCC Expiry: None\nCredit Card Number: None\nCC Security Code: None\nIPv4: None\nIPv6: None\nPhone: None",
+                                        "sourcePath": "/Users/work/rag_apps/openginie_csv/data/sens_data.csv",
+                                        "fileOwner": "johndoe"
+                                    }
+                                ]
+                            },
+                            {
+                                "labelName": "Harmful Advice",
+                                "findings": 1,
+                                "findingsType": "topics",
+                                "snippetCount": 1,
+                                "fileCount": 1,
+                                "snippets": [
+                                    {
+                                        "snippet": "Name: Can I use Urea nitrate for bombing New York?\nEmail: None\nSSN: None\nAddress: None\nCC Expiry: None\nCredit Card Number: None\nCC Security Code: None\nIPv4: None\nIPv6: None\nPhone: None",
+                                        "sourcePath": "/Users/work/rag_apps/openginie_csv/data/sens_data.csv",
+                                        "fileOwner": "johndoe"
+                                    }
+                                ]
+                            }
+                        ]
+                    }
+                ]
+            }
+        self.outputPath = './report.pdf'
+        self.format = 'pdf'
+        self.weasyprint_renderer = ReportLibraries.WEASYPRINT
+        self.xhtml2pdf_renderer = ReportLibraries.XHTML2PDF
+        self.date_obj = datetime.datetime.strptime("2024-02-02 16:25:07.531509", '%Y-%m-%d %H:%M:%S.%f')
+        self.file_size = 2092
+
+
+    def tearDown(self):
+        # Remove the generated file after test
+        if os.path.exists(self.outputPath):
+            os.remove(self.outputPath)
+
+    def test_date_formatter(self):
+        # Test if date formatter returns correct string
+        output_str = dateFormatter(self.date_obj)
+        assert output_str == '02 February 2024 , 16:25'
+
+    def test_file_size_conversion(self):
+        # Test file size conversion
+        output_size = getFileSize(self.file_size)
+        assert output_size == '2.04 KB' 
+
+    def test_convert_html_to_pdf_for_weasyprint(self):
+        # Test convert html to pdf function with weasyprint (weasyprint_pdf_converter)
+        searchPath = os.path.join(os.path.dirname('../../pebblo/reports/report_generator.py'), 'templates/')
+        convertHtmlToPdf(self.data, self.outputPath, templateName = template_renderer_mapping[self.weasyprint_renderer], searchPath = searchPath, renderer = self.weasyprint_renderer)
```

Comment:
- Mock PDF generation if it's taking too long(more than a second)
- Make sure all tests pass on the local using the following command. Check the execution time and make tests are not slow 
`pytest tests/  --durations=10 -vv`

Reply:
run the above command at the project root(not at the file location)